### PR TITLE
feat(on-function-explicit-return-type): add suggestion

### PR DESCRIPTION
**Pull request by rafaelss95**
Part of #179.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -50,7 +50,7 @@
   },
   "dependencies": {
     "@angular-devkit/schematics": "^12.2.2",
-    "@typescript-eslint/experimental-utils": "^4.29.2",
+    "@typescript-eslint/experimental-utils": "^4.31.0",
```
Comment (rafaelss95) [inline package.json:53]: I forgot to update this package along with others before 😞

File: src/rules/store/on-function-explicit-return-type.ts [hunk 2]
```diff
@@ -16,22 +20,54 @@ export default ESLintUtils.RuleCreator(docsUrl)<Options, MessageIds>({
...
+  sourceCode: Readonly<TSESLint.SourceCode>,
+  fixer: TSESLint.RuleFixer,
+): TSESLint.RuleFix | readonly TSESLint.RuleFix[] {
+  const [firstParam] = params
+  const [lastParam] = params.slice(-1)
+  const previousToken = sourceCode.getTokenBefore(firstParam)
+  const isParenthesized =
+    previousToken && ASTUtils.isOpeningParenToken(previousToken)
+
+  if (isParenthesized) {
```
Comment (rafaelss95) [inline src/rules/store/on-function-explicit-return-type.ts:64]: I tried to use [`ASTUtils.isParenthesized`](https://github.com/typescript-eslint/typescript-eslint/blob/1c1b572c3000d72cfe665b7afbada0ec415e7855/packages/experimental-utils/src/ast-utils/eslint-utils/astUtilities.ts#L113-L116) but it doesn't work for this case, as it only returns `true` if a node has `parenthesis` before **and** after, which isn't necessarily our case, as we can have, for example, `on(action, (a, b) => ...`, so neither param `a`, nor param `b` would receive `true` as a response from this `function`.

File: src/rules/store/on-function-explicit-return-type.ts [hunk 3]
```diff
@@ -16,22 +20,54 @@ export default ESLintUtils.RuleCreator(docsUrl)<Options, MessageIds>({
...
       description: 'On function should have an explicit return type.',
       recommended: 'warn',
+      suggestion: true,
     },
     schema: [],
     messages: {
-      [messageId]:
+      [onFunctionExplicitReturnType]:
         'On functions should have an explicit return type when using arrow functions: `on(action, (state): State => {}`.',
+      [onFunctionExplicitReturnTypeSuggest]: 'Add explicit return type.',
```
Comment (timdeschryver) [inline src/rules/store/on-function-explicit-return-type.ts:29]: I think we should mention that we've chosen `State` as return type, but that the user should update it if needed.
What do you think?

Comment (rafaelss95) [reply src/rules/store/on-function-explicit-return-type.ts:29]: Cool. What's your suggestion?

Comment (timdeschryver) [reply src/rules/store/on-function-explicit-return-type.ts:29]: Ha! that's a good question (and of course it's easy for me to point this out without providing something better) 😅
Something like this?

"Add explicit return type (By default `State`, when the interface is named different you need to manually correct the return type)."

Comment (rafaelss95) [reply src/rules/store/on-function-explicit-return-type.ts:29]: What about this? Feel free to apply the suggestion if you agree :)

```suggestion
      [onFunctionExplicitReturnTypeSuggest]: 'Add explicit return type `State` (if the interface is named differently you need to manually correct the return type).',
```

**Review by timdeschryver — APPROVED** [review]

**Pull request merged**